src/libmongoc/doc/howto/source-install.rst

```diff
@@ -0,0 +1,240 @@
+=============================================================
+
+It is recommended that you install custom-built |mongo-c-driver| libraries in an
+unprivileged filesystem location particular to the user account.
+
+.. tab-set::
+
+  .. tab-item:: macOS
+
+    Unlike other Unix-like systems, macOS does not have a specific directory for
+    user-local package installations, and it is up to the individual to create
+    such directories themselves.
+
+    The choice of directory to use is essentially arbitrary. For per-user
+    installations, the only requirement is that the directory be writeable by
+    the user that wishes to perform and use the installation.
+
+    For the purposes of uniformity with other Unix variants, this guide will
+    lightly recommend using `$HOME/.local` as a user-local installation prefix.
+    This is based on the behavior specified by the `XDG base directory`_
+    specifications and the `systemd file-hierarchy`_ common on Linux and various
+    BSDs, but it is not a standard on other platforms.
+
+  .. tab-item:: Linux & Other Unixes
+
+    On Linux and BSD systems, it is common to use the `$HOME/.local` directory
+    as the prefix for user-specific package installations. This convention
+    originates in the `XDG base directory`_ specification and the
+    `systemd file-hierarchy`_
+
+    Because of its wide-spread use and support in many other tools, this guide
+    recommends using `$HOME/.local` as a user-local installation prefix.
+
+  .. tab-item:: Windows
+
+    On Windows, there exists a dedicated directory for user-local files in
+    :batch:`%UserProfile%\\AppData\\Local`. To reference it, expand the
+    :batch:`%LocalAppData%` environment variable. (**Do not** use the
+    :batch:`%AppData%` environment variable!)
+
+    Despite this directory existing, it has no prescribed structure that suites
+    our purposes. The choice of user-local installation prefix is arbitrary.
+    This guide *strongly discourages* creating additional files and directories
+    directly within the user's home directory.
+
+    Consider using :batch:`%LocalAppData%\\MongoDB` as a prefix for the purposes
+    of manually installed components.
+
+
+.. _source-install.system-prefix:
+
+Selecting a System-Wide Installation Prefix
+===========================================
+
+If you wish to install the |mongo-c-driver| libraries in a directory that is
+visible to all users, there are a few standard options.
+
+.. tab-set::
+
+  .. tab-item:: Linux, macOS, BSD, or Other Unix
+
+    Using an install `$PREFIX` of ``/usr/local/`` is the primary recommendation
+    for all Unix platforms, but this may vary on some obscure systems.
+
+    .. important::
+
+      **DO NOT** use ``/usr/`` nor ``/`` (the root directory) as a prefix: These
+      directories are designed to be carefully managed by the system. The
+      ``/usr/local`` directory is intentionally reserved for the purpose of
+      unmanaged software installation.
```

Comment:
I think this paragraph + the "strongly discouraged" paragraph below for Windows deserve to be marked as "Warning" admonitions.